Client topology installed during authentication can cause NPE in client

### Describe the bug

Same setup as with 13734 but using 15.1.4.Final

1.	The client connects to the VIP.
2.	The VIP responds with a topology update containing the list of real server IPs. The VIP itself is not included in that list.
3.	The client establishes connections to the real server IPs and closes the connection to the VIP.
4.	AuthHandler calls OperationChannel.markAcceptingRequests on closed VIP channel
5.	NPE is thrown

Caused by: java.lang.NullPointerException: Cannot invoke "io.netty.channel.Channel.eventLoop()" because "this.channel" is null
	at org.infinispan.client.hotrod.impl.transport.netty.OperationChannel.markAcceptingRequests(OperationChannel.java:122)
	at org.infinispan.client.hotrod.impl.transport.netty.AuthHandler.lambda$activate$1(AuthHandler.java:86)
	at java.base/java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:831)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:554)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2223)
	at org.infinispan.client.hotrod.impl.transport.netty.OperationDispatcher.handleResponse(OperationDispatcher.java:688)
	at org.infinispan.client.hotrod.impl.transport.netty.OperationDispatcher.handleResponse(OperationDispatcher.java:670)
	at org.infinispan.client.hotrod.impl.transport.netty.HeaderDecoder.decode(HeaderDecoder.java:292)
	at org.infinispan.client.hotrod.impl.transport.netty.HintedReplayingDecoder.callDecode(HintedReplayingDecoder.java:94)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)

### Expected behavior

client statrup finishes without exceptions

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Infinispan version or git rev

15.1.4.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/HeaderDecoder.java
+++ b/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/HeaderDecoder.java
@@ -369,8 +369,12 @@ public class HeaderDecoder extends HintedReplayingDecoder<HeaderDecoder.State> {
          segmentOwners = null;
       }
 
-      dispatcher.updateTopology(cacheName, operation, newTopologyId,
-            addresses, segmentOwners, hashFunctionVersion);
+      // Only update the topology if this channel is fully connected and authenticated
+      OperationChannel operationChannel = channel.attr(OperationChannel.OPERATION_CHANNEL_ATTRIBUTE_KEY).get();
+      if (operationChannel != null && operationChannel.isAcceptingRequests()) {
+         dispatcher.updateTopology(cacheName, operation, newTopologyId,
+               addresses, segmentOwners, hashFunctionVersion);
+      }
    }
 
    private InetSocketAddress[] readTopology(ByteBuf buf) {

--- a/client/hotrod-client/src/test/java/org/infinispan/client/hotrod/PingOnStartupTest.java
+++ b/client/hotrod-client/src/test/java/org/infinispan/client/hotrod/PingOnStartupTest.java
@@ -41,6 +41,8 @@ public class PingOnStartupTest extends MultiHotRodServersTest {
             new InternalRemoteCacheManager(clientBuilder.build())) {
          @Override
          public void call() {
+            // The topology isn't installed until the first operation now
+            assertEquals(0, rcm.getCache().size());
             OperationDispatcher dispatcher = rcm.getOperationDispatcher();
             eventuallyEquals(2, () -> dispatcher.getServers().size());
          }

